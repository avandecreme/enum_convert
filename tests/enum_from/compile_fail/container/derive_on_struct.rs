use enum_convert::EnumFrom;

enum Source {
    Unit,
}

#[derive(EnumFrom)] // Should only work on enums
#[enum_from(Source)]
struct Target {
    field: i32,
}

fn main() {}
