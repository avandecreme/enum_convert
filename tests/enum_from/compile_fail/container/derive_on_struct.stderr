error: proc-macro derive panicked
 --> tests/enum_from/compile_fail/container/derive_on_struct.rs:7:10
  |
7 | #[derive(EnumFrom)] // Should only work on enums
  |          ^^^^^^^^
  |
  = help: message: EnumFrom can only be derived for enums
